### PR TITLE
lock down a specific node version

### PR DESCRIPTION
**Pull request by faithngetich**
#### What does this PR do?
- Lock our project to a more recent version of node that generates a package.json
#### Description of Task to be completed?
- Lock our project to a more recent version of node that generates a package.json
#### How should this be manually tested?
- Clone the repo
- Checkout into this branch
- Run `npm-install`
If your node version is less than the one specified you should get an error
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[148842665](https://www.pivotaltracker.com/n/projects/742821/stories/148842665  )

#### Screenshots (if appropriate)
- N/A
#### Questions: 
- N/A

### CODE REVIEW

**Comment by sigu** [comment]
@faithngetich, 
you might have mistakenly checked in some files that are not relevant to this PR. Maybe you did `git add -A` 😛 If you need help on how to remove them from the PR let me know
2. `npm` now comes with a `package.lock` file for locking the versions, could we consider this too? (For us to do this then we need npm version > 5 🤔 )

**Comment by nisevi** [comment]
Hi there @faithngetich :+1: 

As @sigu said the first thing that you will have to do is to remove the last commit you have done with this command, while in your branch `148842665_node_version_lockdown`:

`git reset HEAD~`

After doing that locally you will have again all changes lets say in `red` :+1 :
image

When you reach this point I recommend you [this reading](https://github.com/AgileVentures/LocalSupport/blob/develop/CONTRIBUTING.md)  specially [this part](https://github.com/AgileVentures/LocalSupport/blob/develop/CONTRIBUTING.md#acceptance-testing--caching)

After your reading :+1: , you will find out that running this command:

`rake vcr_billy_caches:reset`

or maybe you will have to add:

`bundle exec rake vcr_billy_caches:reset`

it will clean the unnecessary files that were generated here `features/req_cache/**`

Once you have only the files that you have modified you can add them for a next commit, but you will face a new problem while trying to push your changes, since we removed HEAD locally with `git reset HEAD~` your local branch now is not synchronized with your branch on the cloud, so you will need to force it, here are the steps you will need to follow:

```
git add .nvmrc
git add bower.json
git add docs/installation/osx.md
git add package-lock.json
git add package.json
git add yarn.lock
```

or

`git add .nvmrc bower.json docs/installation/osx.md package-lock.json package.json yarn.lock`

no idea if they are all the files you wanted to add but since there are more than 500 I'm not checking all of them :P ...

After you add the respective files:

`git commit -m "lock down a specific node version"`

Here if you try to push it will fail as I mentioned before, so you will have to:

`git push origin 148842665_node_version_lockdown -f`

the `-f` flag will force the push and will override your previous HEAD so it will be a clean start :+1:

**Comment by nisevi** [comment]
Some tools that I use are:

- [git-gui](https://git-scm.com/docs/git-gui)
    - `sudo apt-get install git-gui`
- [gitk](https://git-scm.com/docs/gitk)
    - `sudo apt-get install gitk`

Here you have other [GUI Clients](https://git-scm.com/download/gui/linux) to take a look if you want, but I just use the `built-in` tools that come with `git`.

**Comment by sigu** [comment]
Very nice explanation @nisevi  👏

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+node
```
Comment (sigu) [inline .nvmrc]: We can lock this to the latest LTS (`8.9.4`)

File: bower.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "LocalSupport",
+  "name": "localsupport",
```
Comment (sigu) [inline bower.json]: Seems you changed this cause of the warning? 😄

File: docs/installation/osx.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ Type `psql` into command line. Then you should see this:
 Next to that type `ALTER ROLE "postgres" WITH CREATEDB`.  (You may need to create this role, see [issues](issues.md) for help.)
 1. `git checkout develop`.
 1. Run `bundle install` to get the gems.
-1. Run `npm install` to get the javascript dependencies.
+1. Run `yarn install` to get the javascript dependencies.
```
Comment (sigu) [inline docs/installation/osx.md]: Change this back to `npm` so we use `package-lock.json` to lock package versions

File: package.json [hunk 4]
```diff
@@ -22,6 +22,7 @@
   },
   "homepage": "https://github.com/AgileVentures/LocalSupport",
   "dependencies": {
-    "bower": "^1.3.12"
+    "bower": "^1.3.12",
```
Comment (faithngetich) [inline package.json]: Hi @nisevi , what do you think about this change? Does it solve for our current problem?

**Comment by sigu** [comment]
Worked with @faithngetich on this, won't merg it in. We need a third eye 😛

File: package.json (hunk 4 above)
Comment (nisevi) [reply package.json]: @faithngetich nice work :+1:

**Comment by sigu** [comment]
Need to update installation notes to suggest to the user to have nvm in their machine and use it

**Comment by tansaku** [comment]
so I think what we probably need to do with this is check the branch out on a fresh instance of c9 (or other clean computer that never had LocalSupport on it) and check the install process works fine there -

**Comment by tansaku** [comment]
we were just looking at this - think we need to include package-lock.json in this, and also update our c9 (and other instructions) to say to install a particular version of node ...

**Comment by tansaku** [comment]
So the outstanding question about this PR is whether a separate check-version script is the way to go here - is that what other projects are doing?

There seem to be a few ways to do it:

https://www.google.com/search?q=enforce+node+version

apologies if maybe we should have just pulled this in and iterated ... :-/

maybe any way to do this is fine, if there is no standard ...

**Comment by tansaku** [comment]
I did manage to trigger the check locally when using node v7.4.0:

```
Error: Required node version >=8.9.4 not satisfied with current version v7.4.0.
    at Object.<anonymous> (/Users/tansaku/Documents/GitHub/AgileVentures/LocalSupport/check-version.js:6:9)
```
so maybe we are all good, and we just need to make sure that all our install docs match ...

**Pull request merged**